Add user command event types

zhao-oai opened the pull request:
adding new user command event, logic in TUI to render user command events

zhao-oai committed:
feat: add user command event types

zhao-oai committed:
test: update TUI snapshots for user command events

zhao-oai committed:
fix test

zhao-oai committed:
refactor removing usershellcmd event msg type

zhao-oai committed:
stop persisting execcommand eventmsg

zhao-oai commented on codex-rs/core/src/rollout/policy.rs:
```diff
@@ -45,6 +45,9 @@ pub(crate) fn should_persist_event_msg(ev: &EventMsg) -> bool {
         | EventMsg::ExitedReviewMode(_)
         | EventMsg::UndoCompleted(_)
         | EventMsg::TurnAborted(_) => true,
+        EventMsg::ExecCommandBegin(ev) => ev.is_user_shell_command,
```
sure! it's done

pakrym-oai commented on codex-rs/protocol/src/protocol.rs:
```diff
@@ -1258,6 +1262,10 @@ pub enum ExecOutputStream {
+    /// True when this exec was initiated directly by the user (e.g. bang command),
+    /// not by the agent/model. Defaults to false for backwards compatibility.
+    #[serde(default)]
+    pub is_user_shell_command: bool,
```
Do we actually need this on the delta itself of can it be on start ?

pakrym-oai reviewed (changes requested):
is it enought to have is_user_shell_command only on the start?

zhao-oai committed:
undo chatwidget diff

zhao-oai committed:
undoing UI diff

zhao-oai committed:
Revert "undo chatwidget diff"

This reverts commit 5f6983a50f79e0854712a06ac57dc3f0e726c457.

zhao-oai committed:
remove :

zhao-oai committed:
undoing diff

zhao-oai committed:
further undoing diffs

zhao-oai committed:
.

zhao-oai committed:
removing is_user_shell_command

zhao-oai commented on codex-rs/tui/src/snapshots/codex_tui__history_cell__tests__stderr_tail_more_than_five_lines_snapshot.snap:
```diff
@@ -2,7 +2,7 @@
 expression: rendered
 ---
-• Ran seq 1 10 1>&2 && false
+• Ran: seq 1 10 1>&2 && false
```
i thought adding a colon would make it clearer since user commands will render as

`You Ran: ls` instead of `You Ran ls` for example

this is somewhat adjacent though so i'll get this in another PR

zhao-oai committed:
undo diff

zhao-oai commented on codex-rs/tui/src/exec_cell/render.rs:
```diff
@@ -317,7 +345,13 @@ impl ExecCell {
+        let title = if self.is_active() {
+            "Running"
+        } else if call.is_user_shell_command {
+            "You Ran"
```
@codex could you change "You Ran" to "You ran"?

zhao-oai committed:
tui: lowercase user command title

zhao-oai committed:
Wire user command events (#6247)

emit new events from user shell tasks, disable sandboxing for user shell
commands, and present user shell commands to model
